src/main/kotlin/racing/ResultData.kt

```diff
@@ -0,0 +1,14 @@
+package racing
+
+class ResultData {
+
+    private val _resultDataList = mutableListOf<PlayCarList>()
+    val resultDataList: List<PlayCarList>
+        get() = _resultDataList.toList()
+
+    fun saveResultData(playCarList: PlayCarList) {
+
+        val copiedCarList = PlayCarList(playCarList.carList.map { it.copy() })
+        _resultDataList.add(copiedCarList)
+    }
+}
```

Comment:
이 부분을 많이 고민하신 거 같네요. 👏🏾
잘 해주셨고, 추가적으로 다른 방법도 남겨드려볼게요.
아래와 같은 방식으로 람다를 넘겨 처리하면 의존성없이 처리할 수도 있습니다.

```suggestion
    // domain
    fun play(playCnt: Int, resultHandler: (Int, PlayCarList) -> Unit) {
        repeat(playCnt) {
            val isMoveList = carMoveLogic.isMoveList(playCarList.carsSize())
            playCarList.moveCars(isMoveList)
            resultHandler(it, playCarList)
        }
    }
    
    // main
    racing.play(inputPlayCnt.toInt()) { index, result ->
        ResultView.showRacingResult(index, result)
    }
    
    // view
    fun showRacingResult(index: Int, result: PlayCarList) {
        println("${index + FIXED_INDEX}회차 경주")
        printRacingResult(result)
        println()
    }
```